### ISSUE TITLE
Migration of a Laravel existing project

### ISSUE THREAD

**Issue by ttnt-1013** [vitessio/vitess#8151]
Hi there,

I'm trying to connect Laravel pod and MySQL on [Vitess operator (101_initial_cluster.yaml)](https://github.com/vitessio/vitess/blob/master/examples/operator/101_initial_cluster.yaml) on my local machine.
When I tried `php artisan migrate`, the syntax error was happened.

**SQLSTATE[HY000]: General error: 1105 unknown error: syntax error at position 55 near 'users_id_primary' (SQL: alter table `users` add primary key `users_id_primary`(`id`))**

The queries were here:
```
CreateUsersTable: create table `users` (`id` char(36) not null, `name` varchar(255) not null, `email` varchar(255) not null, `email_verified_at` timestamp null, `password` varchar(255) not null, `language_settings` varchar(255) null, `avatar` varchar(255) null, `provider` varchar(20) null, `provider_id` varchar(255) null, `access_token` varchar(255) null, `remember_token` varchar(100) null, `created_at` timestamp null, `updated_at` timestamp null) default character set utf8mb4 collate 'utf8mb4_unicode_ci'

CreateUsersTable: alter table `users` add primary key `users_id_primary`(`id`)

...
```

So it means the 2nd query caused the error.
When I checked MySQL directly (by[ pf.sh](https://github.com/vitessio/vitess/blob/master/examples/operator/pf.sh)), I confirmed the first query was succeeded.
What I did't understand was when I tried directly the query that caused the syntax error, it was succeeded. (I did copy & paste this: `alter table `users` add primary key `users_id_primary`(`id`)`)

I guess I should confirm query logs sent to VTGate or Vttablet but how can I see them?
And do you have any idea I should do as next step?

Thank you!

**Comment by systay** [comment]
👋 @ttnt-1013 

Thank you so much for reporting this!

This seem to be the same issue as #8149. I'm going to close this one and ask you to follow the progress on the other issue instead.

Your next step is to wait until someone can pick this up and make the parser understand this syntax.

**Comment by shlomi-noach** [comment]
For immediate relief, consider changing the query from:
```sql
alter table `users` add primary key `users_id_primary`(`id`)
```
(technically invalid syntax), to:
```sql
alter table `users` add primary key (`id`)
```

**Comment by ttnt-1013** [comment]
@shlomi-noach 
Thank you for the suggestion.
But Laravel generates the query on background and as far as I know, there is no way to modify query. (And I don't know why Laravel add `users_id_primary`...).
Can you guess why `users_id_primary` was added?
If you have any ideas for this, please kindly let me know.

**Comment by shlomi-noach** [comment]
> Can you guess why users_id_primary was added?

My only guess is that this is how the framework found it easiest to generate the query, and seeing that MySQL allowed it, that became the standard for the framework. We're looking into supporting this in Vitess.

**Comment by ttnt-1013** [comment]
I see.
When are you guys actually going to support this? 
If you don't have any schedule to do, I might need to find some trick to deal with this.

**Comment by shlomi-noach** [comment]
> When are you guys actually going to support this?
We're looking into it. I can't commit to a specific schedule.

**Comment by ttnt-1013** [comment]
Do you know which part of Vitess does return the syntax error?
I just thought if Vitess does only pass SQL to MySQL and the result to app, the error won't be happened.

**Comment by shlomi-noach** [comment]
> Do you know which part of Vitess does return the syntax error?

We do.

> I just thought if Vitess does only pass SQL to MySQL and the result to app, the error won't be happened.

You are absolutely correct. We're looking into fixing it in Vitess.

**Comment by ttnt-1013** [comment]
Thanks again!
I just wondered if I could avoid this issue on Vitess of current latest version.
Let me know if I could give some info that might help your works. :)

**Comment by shlomi-noach** [comment]
Fixed in #8188 ; it will take some time until we push to PlanetScale.
